lib/policyEvaluator/utils/wildcards.js

```diff
@@ -0,0 +1,62 @@
+'use strict'; // eslint-disable-line strict
+
+
+const wildcards = {};
+
+// * represents any combo of characters
+// ? represents any single character
+
+// TODO: Note that there are special rules for * in Principal.
+// Handle when working with bucket policies.
+
+
+/**
+ * Converts string into a string that has all regEx characters escaped except
+ * for those needed to check for AWS wildcards.  Converted string can then
+ * be used for a regEx comparison.
+ * @param {string} string - any input string
+ * @return {string} converted string
+ */
+wildcards.handleWildcards = string => {
+    // Replace all '*' with '.*' (allow any combo of letters)
+    // and all '?' with '.{1}' (allow for any one character)
+    // If *, ? or $ are enclosed in ${}, keep literal *, ?, or $
+    function characterMap(char) {
+        const map = {
+            '\\*': '.*',
```

Comment:
I would need a second opinion on this but I think should we convert all greedy regex checks to lazy. It would be eating a ton of CPU cycles if not.
So `*` becomes `*?`, unless you are matching for a `*` character of course.